packages/components/src/icon/index.js

```diff
@@ -61,6 +61,8 @@ function Icon( { icon = null, size, ...additionalProps } ) {
 
 	if ( isValidElement( icon ) ) {
 		return cloneElement( icon, {
+			height: iconSize,
+			width: iconSize,
```

Comment:
I'm on the fence here, this means that "icons as elements" should have support for "width"/"height" props. If for some reason, these props are used for something else, it could break. 

I'd appreciate thoughts from other persons here. @jorgefilipecosta @gziolo @ItsJonQ

Reply:
@aktasfatih What if we check for `icon.type === SVG` (SVG from primitives) in the condition above, would it fix the issue?

Reply:
If the icon is created using the ```SVG``` tag or ```<svg />```, it already gets caught in the if statement that is right before it. That totally works both for mobile and web. Since all the icons are created by exporting an inline ```<svg />``` in javascript, there are no problems related to the inline SVGs. The corresponding SVG check is below.
```
 if ( icon && ( icon.type === 'svg' || icon.type === SVG ) ) { 
 	const appliedProps = { 
 		width: iconSize, 
 		height: iconSize, 
 		...icon.props, 
 		...additionalProps, 
 	}; 
  
 	return <SVG { ...appliedProps } />; 
 } 
```
The problem arises when plugins try to import an ```Icon.svg``` file directly, using something like ```import PluginIcon from "./myicon.svg";```. React doesn't directly recognize this as an ```<svg />``` or our ```Wordpress/SVG``` object. It reads the whole file as a string. So, people benefit from modules like ```react-scripts```, or ```svg-react-loader``` to configure webpack to load it as a component. However, the type ends up having different such as ```type = PluginIcon```. So, it skips the SVG check and goes to the isValidElement.